[14.0] Correção para carregar o CST corretamente , qdo operação de entrada.

crsilveira opened the pull request.

ATSTI committed:
Correção para carregar o CST corretamente , qdo operação de entrada.

antoniospneto commented on l10n_br_fiscal/models/tax_definition.py:
```diff
@@ -364,3 +364,8 @@ def _onchange_tax_id(self):
+
+    @api.onchange("tax_group_id")
+    def _onchange_tax_group_id(self):
+        if self.tax_group_id:
```
@crsilveira 

Só tem um problema, nem sempre o modelo **tex_definition** irá ter um **fiscal_operation_line_id**
é o caso quando você tenta configurar os impostos padrões na lá na configuração da empresa, o erro de validação acontece:

![image](https://user-images.githubusercontent.com/634278/152450567-61b1555d-7d76-444b-97f8-b644597a91f5.png)

Precisa adicionar uma verificação se o fiscal_operation_line_id existe antes.

ATSTI committed:
Ajustando pra quando não tem um fiscal_operation_line_id.

This reverts commit da60a23232c0fa904c7309918b12d44e977fcdf6.

Ajustando pra quando não tem um fiscal_operation_line_id.

ajusta última linha.

Revert "ajusta última linha."

This reverts commit da60a23232c0fa904c7309918b12d44e977fcdf6.

Revert "Ajustando pra quando não tem um fiscal_operation_line_id."

This reverts commit 2fd362e7d4bde094673d6874621bb7205ef9634a.

Ajustando pra quando não tem um fiscal_operation_line_id.

antoniospneto reviewed (approved):
Ai só acho que poderia melhorar ali o histórico de commits, fazer um squash com o git para unifica-los, isso ajuda a manter a branch principal limpa

ATSTI committed:
Adicionado o dominio da posição fiscal, permitindo configurar a operação conforme a posição fiscal do cliente, ou do pedido.

mileo commented:
> Ai só acho que poderia melhorar ali o histórico de commits, fazer um squash com o git para unifica-los, isso ajuda a manter a branch principal limpa

Para uma primeira contribuição acho isso muito complicado. @crsilveira vc sabe fazer?

```
git rebase -i --autosquash HEAD~4
```

Vai abrir o editor de texto (4 é o numero de commits que vc quer ver no editor)

E ai vc altera os dois últimos para squash, que ele junta com o primeiro.

![image](https://user-images.githubusercontent.com/1975978/164893184-94b0a122-24ae-4280-950a-66274fe35d2a.png)

crsilveira committed:
Merge branch 'OCA:14.0' into 14.0

ATSTI committed:
Carregando o ind_final do partner_id

crsilveira committed:
Merge branch 'OCA:14.0' into 14.0

ATSTI committed:
Merge remote-tracking branch 'upstream/14.0-ocabot-merge-pr-1966-by-renatonlima-bump-major' into 14.0

crsilveira committed:
Merge branch 'OCA:14.0' into 14.0

ATSTI committed:
Quando a nota já existe na receita dá erro q não tem o attribute protocolo.

crsilveira committed:
Merge branch 'OCA:14.0' into 14.0
